foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java

```diff
@@ -373,20 +418,26 @@ void embedding() {
     assertThat(embeddingData.getEmbedding())
         .isNotNull()
         .isNotEmpty()
-        .isEqualTo(new float[] {0.0f, 3.4028235E38f, 1.4E-45f, 1.23f, -4.56f});
+        .containsExactly(
+            new BigDecimal("0.0"),
+            new BigDecimal("3.4028235E+38"),
+            new BigDecimal("1.4E-45"),
+            new BigDecimal("1.23"),
+            new BigDecimal("-4.56"));
```

Comment:
(Major/Comment)

Instead of `float[]` we have `List<BigDecimal>` for embedding vectors. This needs finetuning.